cmd/krew/cmd/update.go

```diff
@@ -41,12 +57,147 @@ Remarks:
 	RunE: ensureIndexUpdated,
 }
 
+func retrieveUpdatedPluginList() ([]string, error) {
+	modifiedFiles, err := gitutil.ListModifiedFiles(constants.IndexURI, paths.IndexPluginsPath())
+	if err != nil {
+		return []string{}, err
+	}
+
+	var plugins []string
+	for _, f := range modifiedFiles {
+		filename := filepath.Base(f)
+		extension := filepath.Ext(filename)
+		name := filename[0 : len(filename)-len(extension)]
+		plugins = append(plugins, name)
+	}
+
+	return plugins, nil
+}
+
+func retrievePluginNameVersionMap(names []string) map[string]string {
+	m := make(map[string]string, len(names))
+	for _, n := range names {
+		plugin, err := indexscanner.LoadPluginByName(paths.IndexPluginsPath(), n)
+		if err != nil {
+			continue
+		}
+
+		m[n] = plugin.Spec.Version
+	}
+
+	return m
+}
+
+func retrieveInstalledPluginMap() (map[string]bool, error) {
+	plugins, err := installation.ListInstalledPlugins(paths.InstallReceiptsPath())
+	if err != nil {
+		return map[string]bool{}, err
+	}
+	m := make(map[string]bool, len(plugins))
+	for name := range plugins {
+		m[name] = true
+	}
+
+	return m, nil
+}
+
+func filterAndSortUpdatedPlugins(old, updated map[string]string, installed map[string]bool) ([]newPlugin, []newVersionPlugin) {
+	var newPluginList []newPlugin
+	var newVersionList []newVersionPlugin
```

Comment:
something I noticed throughout your coding style is how you prefix things with their types (e.g. xxxMap xxxList). let's avoid that. :)

you can simply use `newPlugins`, `updatedPlugins` here.